### ISSUE TITLE
Add support for texture maps on surfaces/meshes

### ISSUE THREAD

**Issue by jni** [napari/napari#5568]
## 🚀 Feature

Emmanuel Reynaud has some gorgeous surface datasets with texture maps e.g. [here](https://sketchfab.com/therlab) that I think we should be able to display in napari. Plus the corals would be a great fit for the island theme of napari. 😃 VisPy meshes support texture map data, see example [here](https://vispy.org/gallery/scene/mesh_texture.html#sphx-glr-gallery-scene-mesh-texture-py).

Emmanuel has also hinted in (forum not found) (**Edit:** it was [Twitter!](https://twitter.com/jnuneziglesias/status/1579795263190769665)) that they have explored better ways to visually map texture information onto meshes from photogrammetry, but I can't find the references. I also can't seem to find his GH!

Anyway, I suggest we first add basic support (e.g. with a `texture_map=` kwarg to the surface layer), then worry about improving it later.

I've also had conversations with Emmanuel that they have textured meshes + volumetric datasets registered on the same space, which I think would be a killer app for napari.

**Comment by aganders3** [comment]
I'm interested in this feature as well and may be able to take it on.

> they have explored better ways to visually map texture information onto meshes from photogrammetry

Do you mean trying to infer texture coordinates rather than supply them? To be clear for this feature I think you mean adding support for textures with existing coordinates.

Similarly, I noticed the [GIfTI](https://surfer.nmr.mgh.harvard.edu/fswiki/GIfTI) format includes colors for each vertex, where napari expects a scalar value for indexing a color map. This is likewise already supported in vispy and shouldn't be too difficult to add in napari.

**Comment by jni** [comment]
> Do you mean trying to infer texture coordinates rather than supply them?

Yes, sorry, I misinterpreted that tweet 😅

> To be clear for this feature I think you mean adding support for textures with existing coordinates.

💯 

> I'm interested in this feature as well and may be able to take it on.

That would be fantastic!

> Similarly, I noticed the GIfTI format includes colors for each vertex, where napari expects a scalar value for indexing a color map. This is likewise already supported in vispy and shouldn't be too difficult to add in napari.

Also great!

**Comment by aganders3** [comment]
I made some progress on this and hope to open a PR next week.

One thing I noticed is that some of the sample data from The R Lab uses multiple textures - note there are some gray spots on the model in my screenshot. I think supporting individual textures is a good start, but we can look at adding that later as I think it may require changes in vispy as well. I also need to learn more about obj and mtl files.

<img width="1148" alt="Screenshot 2023-03-03 at 11 32 33 AM" src="https://user-images.githubusercontent.com/1231828/222785829-09f5507b-2439-4166-899b-00c5e316fa29.png">
